### ISSUE TITLE
Investigate Playwright test unreliability

### ISSUE THREAD

**Issue by kyranjamie** [blockstack/blockstack-app#3919]
More and more I'm seeing random test failures. Let's investigate and try and get to the bottom of why this might be happening.

- Are they temporary API failures? If so, how can we catch/identify?
- Are the tests written poorly, and encountering race conditions?
- Can we use retry logic to reduce occurrences of this happening? (Least favourite option, would be improve reliability other ways)

**Comment by pete-watters** [comment]
Merging issue https://github.com/leather-wallet/extension/issues/4089 here. 

[From](https://github.com/leather-wallet/extension/issues/4089#issue-1832831327)
Our Playwright tests are often failing at the Post run action/cache step citing This operation was cancelled (when it wasn't manually). Some reports say this might be memory related.

image
[Example run →](https://github.com/hirosystems/wallet/actions/runs/5737188117/job/15548315666?pr=4087)

This is problematic in that otherwise successful tests often fail, requiring developers to rerun failed tests. A huge productivity hit.

This issue to to investigate the problem, and try some alternative solutions to remedy the problem. Perhaps we can play with cache settings, have more or fewer job matrices etc

**Comment by pete-watters** [comment]
I've been hitting this issue again on https://github.com/leather-wallet/extension/pull/4655. 

The problematic shards seems to be shard `3` + `5`. Shard `3` generally fails in CI on first run but subsequently passes.

**Comment by pete-watters** [comment]
In working on containers, I noticed that often the unreliability seems to come from when we open the network selector to set testnet for tests. 

I saw that sometimes the network is unavailable for selection, sometimes for `429` errors but sometimes for other reasons. I refactored some of the tests for `send-stx` [here](https://github.com/leather-wallet/extension/pull/4655/commits/f3e129d2be4f1bd73d4e1fee582483b5786fc977) and can return to this soon.

I believe a slight change in how we test could help improve things. I propose adopting a bit of the Kent Dodds [Write fewer, longer tests](https://kentcdodds.com/blog/write-fewer-longer-tests) approach where we don't repeat actions over and over again.

For example, for send stx tests on testnet, we could login and select testnet once and then do all the testing we need to the form in one test, using navigation to reset the form.  So rather than try and login and set the testnet network on every single test we do it just once.
